compiler/rustc_target/src/spec/sbf_solana_solana.rs

```diff
@@ -0,0 +1,64 @@
+use crate::abi::Endian;
+use super::{LinkerFlavor, PanicStrategy, Target, TargetOptions, LldFlavor};
+use std::{collections::BTreeMap};
+
+pub fn target() -> Target {
```

Comment:
If we're planning on bpf and sbf being 100% compatible for the foreseeable future, we should consider using shared target and ABI modules (like bpfel-unknown-none and bpfeb-unknown-none and many other targets do) to reduce copypasta when we make changes.

Reply:
My hope is that this duality of targets will be short lived.  As soon as we update solana-labs/solana to use sbf-solana-solana, we can remove bpfel-unknown-unknown completely. There's no backward compatibility in the sense that we need to preserve the two targets in future releases of bpf-tools (later sbf-tools). The older solana releases will always use the older releases of bpf-tools, which are never updated (instead new versions released). However, even in a short term it's better to minimize the duplication as it's difficult to control the possible divergence of the two targets. I agree with you it's better to use a shared target. I'll fix this.

Reply:
> There's no backward compatibility in the sense that we need to preserve the two targets in future releases of bpf-tools (later sbf-tools). The older solana releases will always use the older releases of bpf-tools, which are never updated (instead new versions released). 

What about people who have stuff like `#[cfg(target_arch = "bpf")]` in their code? Are we ok breaking that when they update toolchain?

Reply:
Good question. I don’t know. Since there are non-solana BPF target arch in rust, it’s probably wrong for on-chain program source to conditionally compile on bpf target_arch. Maybe it’s better to break their code so they update to sbf, when the target bpfel-unknown-unknown is no longer available? The reason we do this rename is to break with BPF, right? I don’t know what’s the correct way to do this